### ISSUE TITLE
Opening a DNG converted from a EOS 90D CR3 File Errors

### ISSUE THREAD

**Issue by juststur** [darktable-org/darktable#4055]
**Describe the bug**

When opening a DNG file converted from an EOS 90D CR3 with Adobe Digital Negative Converter, the error "`Canon Canon EOS 90D' color matrix not found!" will display at the bottom middle of the screen as seen in this capture:
![image](https://user-images.githubusercontent.com/36450068/71943269-0f4c5800-318e-11ea-8938-f4355558120d.png)

**To Reproduce**

Load DNGs converted by Adobe Digital Negative Converter.  Double click image from Lighttable so it opens in darkroom, message "`Canon Canon EOS 90D' color matrix not found!" will display at bottom of screen.

**Expected behavior**

No error will display.

**Screenshots**

**Platform (please complete the following information):**
 - Darktable Version: 
 - OS: 
 - 
 - 

**Additional context**

**Comment by juststur** [comment]
Test DNG https://we.tl/t-CHY8oGCTft

**Comment by guillaume1410** [comment]
Quite strange I don't have the issue on my side, even with your football dng.
What versions are you using ? I use adobe digital negative converter 12.1.0.351 and darktable 3.0.0 (not a rc version)

**Comment by juststur** [comment]
I have the same software versions.  Now that I think about it, the XMP was likely created initially by reading in a filename.XMP file created by another RAW editor (been playing around with others that handle CR3 files naively to see if I wanted to switch programs till darktable was updated).  I don't recall which off hand...maybe ACDSee Photo Studio Ultimate.  Attaching the non darktable XMP and the XMP from darktable.

[xmpfiles.zip](https://github.com/darktable-org/darktable/files/4033707/xmpfiles.zip)

**Comment by juststur** [comment]
I had to grab the non darktable XMP from backup...I had cleared the folder and the error still displays for the images.  The non darktable XMP does contain additional metadata about the EOS 90D that darktable seems to ignore (I'd call that silently loosing data).

**Comment by jenshannoschwalm** [comment]
The message you see results from a color matrix not read/imported from the dng file. An xmp file can contain pretty much everything, so dt might bee "confused". 

Please try:
1. re-read the exif data from within dt? (you find it under the metadata tab in selected image[s])
2. open the image in darkroom

After doing so, does the message still appear?

If it does, please 
1. remove the image from dt database
2. delete the existing xmp file
3. reimport the image into dt

does the problem persist?

**Comment by ptilopteri** [comment]
> I had to grab the non darktable XMP from backup...I had cleared the folder and the error still displays for the images. The non darktable XMP does contain additional metadata about the EOS 90D that darktable seems to ignore (I'd call that silently loosing data).

converting to DNG would definitely be called "silently loosing data".

**Comment by jenshannoschwalm** [comment]
I think i understand what is happening but i do not understand how to fix because i don't know that part of the code at all. I see two important things here and they are absolutely **not** related to the title of this issue. 

1. When you have the "external" xmp file existing besides the image file it is "analysed" by `dt_lightroom_import` when a) importing or b) after discarding the history and reopening in darkroom. Somehow a wrong input color profile is added to the history, so in colorin  `commit_params` looks for DT_COLORSPACE_STANDARD_MATRIX which is not available and thus the error message appears. Also the working profile is set to linearREC709 which also seems to be not correct.

2. Also try to load the given "external" sidecar file -- surprise, surprise -- i have the basecurve and again the wrong input color profile added to history. And - for me an even larger surprise - the console shows a history V1->V2 on the fly conversion!!!

@TurboGit i guess this is something for you when looking at the source history ... BTW this might be another origin of iop_order problems that as far as i remember have never been discussed yet. And, yes, it seems to be a real bug. And, the title of this issue seems to be very misleading.

**Comment by juststur** [comment]
I was able to clear the error.  I believe the sequence of events that lead to the error message was as follows.
1) Use an app that supports CR3 RAW files and writes XMP sidecar files for the loaded RAW files
2) Use darktable to read in DNG files (converted from the CR3) whch are located in the same directory as the original CR3 files.
3) darktable loaded the XMPs in the directory of <file_name>.xmp and merged the data from the two different RAW file types

I removed the DNGs from darktable, removed all the XMP sidecars, opened the DNG files and do not have the errors any longer.

**Comment by jenshannoschwalm** [comment]
Yes and sure. Thats of course the thing you observe and what i described in a more technical way. The point is how the second/wrong sidecar file is interpreted by the lightroom-importer.

**Comment by jenshannoschwalm** [comment]
I don't see the lines but i see very strong artefacts. The are related to values going vastly outside correct values. You should watch out for one of aurelienpierres tutorials about how to use filmic in a better way. I really don't concider this a bug...

**Comment by aurelienpierre** [comment]
Same here with DNG from a scanner and some TIFFs from the same scanner.

**Comment by daniel-lucio** [comment]
I don't have this problem using 3.2.1.
